osu.Game/Storyboards/Drawables/DrawableStoryboardSample.cs

```diff
@@ -21,11 +21,29 @@ public class DrawableStoryboardSample : PausableSkinnableSound
 
         public override bool RemoveWhenNotAlive => false;
 
+        private readonly IBindable<bool> samplePlaybackDisabled;
+
         public DrawableStoryboardSample(StoryboardSampleInfo sampleInfo)
             : base(sampleInfo)
         {
             this.sampleInfo = sampleInfo;
             LifetimeStart = sampleInfo.StartTime;
+
+            samplePlaybackDisabled = SamplePlaybackDisabled.GetBoundCopy();
+        }
+
+        [BackgroundDependencyLoader(true)]
+        private void load()
+        {
+            samplePlaybackDisabled.BindValueChanged(disabled =>
+            {
+                if (!RequestedPlaying) return;
+
+                // Since storyboard samples can be very long we want to stop the playback regardless of
+                // whether or not the sample is looping or not
+                if (disabled.NewValue)
+                    Stop();
```

Comment:
This is also similarly confusing. How does this work? Does this mean that both `PausableSkinnableSound`'s and this value change callback fire? Sounds pretty wrong. The proper way forward is probably abstracting the body of the callback to a virtual method or something.

Reply:
> Why does this make a private copy of a protected bindable owned by this exact same class?

I saw this line in the wiki regarding bindables: _Binding to another class instance's bindable events should be avoided, as it bypasses the reference safety provided by bindables._

Perhaps I misunderstood what it meant, but that code was a result of that.

>This is also similarly confusing. How does this work? Does this mean that both PausableSkinnableSound's and this value change callback fire?

I suppose that would be the case. I wanted to change `PausableSkinnableSound` as little as possible but making a virtual method and overriding it would make sense. I'll try that and see where it goes.

Reply:
Note that it says "to another class *instance*'s". It's not the class *itself* that matters, *who owns the reference* to the bindable matters. In this case both bindables are owned by one and the same reference, so there's no reference/memory leak.

Reply:
Ahh that makes sense, thanks!

Reply:
I've clarified the wording to hopefully be less misleading (says "object/object instance's" now).

Reply:
After working on abstracting the body of the callback to a virtual method, I realized it would be simpler (and likely more readable) to just have an overridable boolean that controlled whether or not non-looping samples would play out to completion in `PausableSkinnableSound`. This boolean could then be overrode in `DrawableStoryboardSample`, instead of needing to assign the `samplePlaybackDisabled.BindValueChanged` callback to a virtual method and overriding that.